iceoryx_utils/include/iceoryx_utils/internal/concurrent/sofi.inl

```diff
@@ -139,46 +139,40 @@ inline bool SoFi<ValueType, CapacityValue>::popIf(ValueType& valueOut, const Ver
 template <class ValueType, uint32_t CapacityValue>
 bool SoFi<ValueType, CapacityValue>::push(const ValueType& valueOut, ValueType& f_paramOut_r) noexcept
 {
-    bool hasOverflow = false;
+    constexpr bool OVERFLOW{false};
 
     uint64_t currentWritePosition = m_writePosition.load(std::memory_order_relaxed);
     uint64_t nextWritePosition = currentWritePosition + 1U;
 
+    m_data[static_cast<int32_t>(currentWritePosition) % m_size] = valueOut;
```

Comment:
We should rename valueOut and paramOut to something more fitting like: inValue, outValue otherwise we are confusing people like me who are just wondering why the out value is written into the SoFi

Reply:
should probably be done in a separate PR